migrator/migration.py

```diff
@@ -29,6 +29,7 @@ def __init__(self, route: CdnRoute, session, client):
         self._external_domain_broker_service_instance = None
         self._space_id = None
         self._org_id = None
+        self._service_plan_visibility_id = None
```

Comment:
why is this attribute protected? If it's following the pattern of the other protected attributes, it should have a public interface, like
```
@property
def service_plan_visibility_id(self):
    ...
```

Reply:
Ah good catch, thanks!  🤔 I am not sure that this needs to be protected now that you call attention to it.  Furthermore, there's no other method to define here to retrieve the ID if it doesn't exist for a couple of reasons:

- There wouldn't be an ID if there's no service plan enabled
- There's no way to retrieve the info without the ID itself, which is returned at the time the service plan is enabled for an org

In other words, there is no underlying method to define to retrieve a value like for the Space ID or Org ID; there isn't a way that I can see to tie information back via CAPI to get back to the service plan visibility, at least not for V2 using the Python client here.  With direct V3 API calls you might be able to stitch things together, but even that's not a clear path.

Does this make sense, or am I missing anything in not keeping this protected?

Reply:
the reason I did this with the others was for lazy-loading. The public `@property` exposes the value, and initializes the value when it's unknown, and the protected field is just where it's stored.  I like this pattern because it means we only hit the CF api when needed, and we know the value will always be initialized when we need it.